### PR TITLE
PR:  gh#149 : Adding changes for menu driven gtest tests

### PR DESCRIPTION
**Pull request by kanjoe24**
Changes for menu driven gtest tests

### CODE REVIEW

File: src/cpp_source/ut_gtest.cpp [hunk 1]
```diff
@@ -18,8 +18,24 @@
 */
 
 #include <ut.h>
+#include <ut_log.h>
 #include <ut_internal.h>
 
+#include <iomanip>
+
+static TestMode_t  gTestMode;
+#define _(x) x
```
Comment (Ulrond) [inline src/cpp_source/ut_gtest.cpp:27]: think you should call this something more sensible, rather than just `_`

**Review by Ulrond — CHANGES_REQUESTED** [review]
minor change

**Review by Ulrond — APPROVED** [review]
looks good, but can we have a demo before we merge.

**Comment by kanjoe24** [comment]
results for ut-core release script:

```shell
./release-test-script-ut-core.sh -t feature/gh149-menu-system-cpp-test
UT_CORE_BRANCH_NAME = feature/gh149-menu-system-cpp-test
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==========================================================
RESULTS for ubuntu 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh149-menu-system-cpp-test
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================

```  

Results for platform release script with commands ./build_ut.sh for linux and ./build_ut.sh TARGET=arm for arm

```shell
./release-test-script-platform.sh -t feature/gh149-menu-system-cpp-test
UT_CORE_BRANCH_NAME = feature/gh149-menu-system-cpp-test
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==========================================================
RESULTS for ubuntu 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-ubuntu
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-VM-SYNC
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
ut/ut-core/framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-dunfell_arm
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-dunfell_linux
ut/bin/hal_test exists. PASS
CURL static lib does not exist. PASS 
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-kirkstone_arm
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh149-menu-system-cpp-test
            . PASS
/home/jpn323/workspace/gh149-menu-system-cpp-test/scripts/rdk-halif-hdmi_cec-kirkstone_linux
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
jpn323@janus ~/workspace/gh149-menu-system-cpp-test/scripts (feature/gh149-menu-system-cpp-test)$ 

```

**Review by Ulrond — APPROVED** [review]
go for it

**Pull request merged**